src/python/pants/option/global_options.py

```diff
@@ -1051,6 +1058,46 @@ class BootstrapOptions:
             """
         ),
     )
+    child_process_max_memory_usage = MemorySizeOption(
+        "--child-process-max-memory-usage",
+        advanced=True,
+        default=DEFAULT_EXECUTION_OPTIONS.child_process_max_memory_usage,
+        default_help_repr="1GiB",
+        help=softwrap(
+            """
+            The maximum memory usage for all child processes.
+
+            The value is participates in precomputing the pool size of child processes used by
+            Pantsd. A high value would result in a high number of child processes spawned,
+            potentially overconsuming your resources and triggering the OS' OOM killer. A low
+            value would mean a low number of child processes launched and therefore less
+            paralellism for the tasks that need those processes.
+
+            You can suffix with `GiB`, `MiB`, `KiB`, or `B` to indicate the unit, e.g.
+            `2GiB` or `2.12GiB`. A bare number will be in bytes.
+            """
+        ),
+    )
+    child_process_default_memory_usage = MemorySizeOption(
+        "--child-process-default-memory-usage",
+        advanced=True,
+        default=DEFAULT_EXECUTION_OPTIONS.child_process_default_memory_usage,
+        default_help_repr="256MiB",
+        help=softwrap(
+            """
+            The default memory usage for a child processes.
+
+            The value is participates in precomputing the pool size of child processes used by
```

Comment:
Can refer to the other option by name rather than repeating the help here probably.